### DISCUSSION TITLE
Serialise accelerator tasks only on local executor?

### DISCUSSION THREAD

**Discussion by SamStudio8** [nextflow-io/nextflow#3806] [Q&A]
Is there an idiom for setting maxForks to 1 for a particular process but only when the executor is local? As Nextflow does not manage GPU devices (to prevent scheduling processes to the same GPU device), I would like to automatically configure a workflow to serialise GPU tasks for local executors -- without user intervention.

I note that it is [one of the directives that cannot be set dynamically](https://nextflow.io/docs/latest/process.html#dynamic-directives) (so no closures); is there a trick I am missing? Otherwise it would be useful if one could cap the number of `accelerators` in the `executor.$local` config like you can with `cpus` and `memory`, which would have the same effect without having to use maxForks.

**Comment by pditommaso** [comment]
Only for a specific process or for the overall pipeline (ie. run only one task at time)?

**Reply by SamStudio8** [reply]
For any process that needs a GPU. I want to be able to do something like this:
```
process {
    withLabel:gpu {
        maxForks = { process.executor == "local" ? 1 : null }
    }
}
```

**Reply by pditommaso** [reply]
Nope, this cannot be done. You should use a profile, but I guess that is not what you are trying to achieve

**Reply by bentsherman** [reply]
You would have to add another label e.g. `label 'local'` to your local processes so that you can select them using `withLabel`.
